Sources/HummingbirdAuth/Authenticator/AuthRequestContext.swift

```diff
@@ -15,20 +15,43 @@
 import Hummingbird
 import Logging
 import NIOCore
+import NIOConcurrencyHelpers
 
 /// Protocol that all request contexts should conform to if they want to support
 /// authentication middleware
-public protocol AuthRequestContext: RequestContext {
-    /// Login cache
-    var auth: LoginCache { get set }
+public protocol AuthRequestContext<Identity>: RequestContext {
+    associatedtype Identity: Authenticatable
+
+    /// The authenticated identity
+    var auth: AuthContainer<Identity> { get set }
+}
+
+extension AuthRequestContext {
+    public var identity: Identity? {
+        get { auth.identity }
+        nonmutating set { auth.identity = newValue }
+    }
+}
+
+public struct AuthContainer<Identity: Authenticatable>: Sendable {
+    private let _identity: NIOLockedValueBox<Identity?>
```

Comment:
I don't think we need the AuthContainer. This isn't like sessions where we need to be able to access an edited session in SessionMiddleware after running the route. The identity is set by the authentication middleware and passed along the chain and any changes made to it further down do not need to be propagated back.

Reply:
My reasoning was twofold:
- Partly out of consistency with session
- And I think some middleware might benefit from this knowledge on the way back. Although I don't have proof for this of course.

I'll remove it for now - we can add it back later